cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -128,6 +129,7 @@ func Run(s *options.CMServer) error {
 	}
 
 	kubeconfig.ContentConfig.ContentType = s.ContentType
+	kubeconfig.ContentConfig.GroupVersion = &unversioned.GroupVersion{Group: batch.GroupName, Version: "v2alpha1"}
```

Comment:
Iow. just before starting the controller, you do have access to kubeconfig at that point and should be able to create a new kubeclient easily.